pkg/cmd/admin/registry/registry.go

```diff
@@ -207,30 +209,52 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 		if len(remove) > 0 {
 			return kcmdutil.UsageError(cmd, "You may not pass negative labels in selector %q", cfg.Selector)
 		}
-		nodeSelector = valid
+		cfg.nodeSelector = valid
 	}
 
-	image := cfg.ImageTemplate.ExpandOrDie(cfg.Type)
+	if ports, err := app.ContainerPortsFromString(cfg.Ports); err != nil {
+		return err
+	} else {
+		cfg.ports = ports
+	}
 
-	namespace, _, err := f.OpenShiftClientConfig.Namespace()
-	if err != nil {
-		return fmt.Errorf("error getting client: %v", err)
+	if namespace, _, err := f.OpenShiftClientConfig.Namespace(); err != nil {
+		return fmt.Errorf("error getting namespace: %v", err)
+	} else {
+		cfg.namespace = namespace
 	}
-	_, kClient, err := f.Clients()
-	if err != nil {
+
+	if _, kClient, err := f.Clients(); err != nil {
 		return fmt.Errorf("error getting client: %v", err)
+	} else {
+		cfg.kClient = kClient
 	}
 
 	cfg.Action.Bulk.Mapper = clientcmd.ResourceMapper(f)
 	cfg.Action.Out, cfg.Action.ErrOut = out, cmd.Out()
 	cfg.Action.Bulk.Op = configcmd.Create
 
+	return nil
+}
+
+// Validate ensures the options are valid.
+func (c *RegistryConfig) Validate(cmd *cobra.Command, args []string) error {
+	if len(args) > 0 {
```

Comment:
If this is the only thing being checked I don't mind moving it to `Complete`. Especially that `Complete` processes  arguments. `Validate` is rather for making sure the command arguments (not the cli, but rather `RegistryConfig` in your case)  are valid.